scripts/dev/curriculum/_commands/screenshot.ts

```diff
@@ -1,109 +1,32 @@
-import { mkdir, unlink } from "fs/promises";
+import { mkdir } from "fs/promises";
 import { dirname, relative } from "path";
 
 import puppeteer, { Page } from "puppeteer";
-import sharp from "sharp";
 import slugify from "@sindresorhus/slugify";
 
-type Image = {
-  input: string;
-  top: number;
-  bottom: number;
-  left: number;
-};
-
-const combineScreenshots = async (
-  images: Image[],
-  outputFileName: string,
-  width: number,
-) => {
-  if (images.length > 0) {
-    const totalHeight = images[images.length - 1]!.bottom;
-
-    await sharp({
-      create: {
-        width: width,
-        height: totalHeight,
-        channels: 4,
-        background: { r: 0, g: 0, b: 0, alpha: 0 },
-      },
-    })
-      .composite(images)
-      .toFile(outputFileName);
-  }
-};
-
 const screenshotPage = async (
   page: Page,
   url: string,
   path: string,
   logOpts: { id: string },
 ) => {
   await page.goto(url, {
-    waitUntil: "domcontentloaded",
+    waitUntil: "networkidle0",
   });
 
-  const viewportHeight = await page.evaluate(() => {
-    return window.innerHeight;
-  });
-  const viewportWidth = await page.evaluate(() => {
-    return window.innerWidth;
+  await page.evaluate(() => {
+    document.querySelector("div[data-testid=cookie-banner]")?.remove();
   });
-  const totalHeight = await page.evaluate(() => {
-    return document.body.scrollHeight;
-  });
-  const sections = Math.ceil(totalHeight / viewportHeight);
-
-  let imgHeight = 0;
-  const images: Image[] = [];
-  for (let i = 0; i < sections; i++) {
-    const sectionPath = `${path}_part${i + 1}.png`;
 
-    const def = {
-      input: sectionPath,
-      top: imgHeight,
-      bottom: Math.min(totalHeight, imgHeight + viewportHeight),
-      left: 0,
-    };
-
-    images.push(def);
-    imgHeight = def.bottom;
-  }
-
-  if (images.length < 1) {
-    return;
-  }
-
-  for (let i = 0; i < images.length; i++) {
-    const sectionPath = images[i]!.input;
-
-    await page.evaluate(
-      (scrollY: number) => window.scrollTo(0, scrollY),
-      i * viewportHeight,
-    );
-
-    console.log(
-      `📸 [${logOpts.id}] temp #${i + 1}: ${relative(
-        process.cwd(),
-        sectionPath,
-      )}`,
-    );
-    await page.screenshot({
-      path: sectionPath,
-      clip: {
-        x: 0,
-        y: i * viewportHeight,
-        width: viewportWidth,
-        height: Math.min(viewportHeight, totalHeight - i * viewportHeight),
-      },
-    });
-  }
+  await page.screenshot({
+    path: path,
+    captureBeyondViewport: true,
```

Comment:
Hmmm I was getting non-full pages captures with this option. Will verify on my machine shortly.

Reply:
Yup this works... no idea why it wasn't working previously 🤷